packages/utils/src/jwk-did-utils.ts

```diff
@@ -112,3 +113,112 @@ export function generateJwkFromVerificationMethod(
   const { publicKeyHex, keyType: extractedType } = extractPublicKeyHex(key)
   return createJWK(keyType, publicKeyHex, keyUse)
 }
+
+const isString = (value: any): value is string => typeof value === 'string'
+
+/**
+ * Validate the JOSE header of a JWT
+ * Only support kid with DID URL
+ * @param header
+ * @param payload
+ * Ref: https://www.w3.org/TR/vc-jose-cose/#using-header-params-claims-key-discovery
+ * Ref: https://www.w3.org/TR/vc-jose-cose/#jose-header-parameters-jwt-claims
+ */
+export const validateHeaderJOSE = (header: any, payload: any) => {
+  // Check for 'alg' (Algorithm - Required)
+  if (!isString(header.alg)) {
+    throw new Error('alg must be present and a string')
+  }
+
+  // Check for 'kid' (Key ID - Required)
+  // https://www.w3.org/TR/vc-jose-cose/#kid
+  if (!isString(header.kid)) {
+    throw new Error('kid must be a string')
+  }
+  const isDidUrlKey = /^did:\w+:/.test(header.kid)
+  if (!isDidUrlKey && header.kid.startsWith('did:')) {
+    throw new Error('kid must be a full DID URL when starting with "did:"')
+  }
+
+  // Check for 'iss' (Issuer - Optional)
+  // https://www.w3.org/TR/vc-jose-cose/#iss
+  if ('iss' in header) {
+    if (!('issuer' in payload)) {
+      throw new Error('issuer must be present in payload when iss is in header')
+    }
+
+    let issuerId = extractIssuer(payload, { removeParameters: true })
+    // Check if 'iss' matches 'issuer.id' or 'issuer' string
+    if (issuerId && header['iss'] !== issuerId) {
+      throw new Error('iss in header does not match issuer value')
+    }
+  }
```

Comment:
Based on this spec: https://www.w3.org/TR/vc-jose-cose/#iss,
I think we should add a type check for the `iss` field before comparing it with the `issuer` in the payload.

```ts
  if ('iss' in header) {

    // Added type checking 
    if (!isString(header.iss)) {
      throw new Error('iss must be a string when present')
    }

    if (!('issuer' in payload)) {
      throw new Error('issuer must be present in payload when iss is in header')
    }

    let issuerId = extractIssuer(payload, { removeParameters: true })
    // Check if 'iss' matches 'issuer.id' or 'issuer' string
    if (issuerId && header['iss'] !== issuerId) {
      throw new Error('iss in header does not match issuer value')
    }
  }
```